ast/desugar/Desugar.cc

```diff
@@ -667,6 +679,41 @@ TreePtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what) {
                         }
                     }
                     result = std::move(res);
+                } else if (send->args.size() == 1 && parser::isa_node<parser::ForwardedArgs>(send->args.back().get())) {
+                    // If the call is forwarding arguments like `foo(...)`
+                    // we desugar it as `foo(*T.unsafe(args), **kargs, &block)`
+                    // which is in turn transformed into a call with splat and block
+                    // as this `callWithSplatAndBlock(:foo, T.unsafe(args).to_a.concat([kargs.to_hash]), block)`
+
+                    auto method =
+                        MK::Literal(loc, core::make_type<core::LiteralType>(core::Symbols::Symbol(), send->method));
+
+                    Send::ARGS_store sendargs;
+                    sendargs.emplace_back(std::move(rec));
+                    sendargs.emplace_back(std::move(method));
+
+                    auto args = MK::Local(loc, core::Names::args());
+                    auto args_unsafe = MK::Unsafe(loc, std::move(args));
```

Comment:
Why are we adding the `T.unsafe` ourselves?

I don't think we should silence the error about splats, and should require the user to write

```ruby
T.unsafe(self).foo(...)
```

to get the error to go away:

https://sorbet.run/#%23%20typed%3A%20true%0Adef%20foo(x%2C%20y)%0Aend%0A%0Axs%20%3D%20T%3A%3AArray%5BInteger%5D.new(%5B1%2C%202%5D)%0A%0Afoo(*xs)%0AT.unsafe(self).foo(*xs)